data_pipeline/utils/pipeline_config.py

```diff
@@ -64,7 +64,7 @@ def replace_env_placeholder(
 
 
 TRUE_VALUES = {'true', '1', 't', 'y', 'yes'}
-FALSE_VALUES = {'false', '0', 'f', 'n', 'no'}
```

Comment:
Personally I think it might still be better to explicitly require a default value to be passed in and raise an issue otherwise.

e.g. something like:

```python
def str_to_bool(value: str, default_value: Optional[bool] = None) -> bool:
    if not value and default_value is not None:
        return default_value
    lower_value = value.lower()
    if lower_value in TRUE_VALUES:
        return True
    if lower_value in FALSE_VALUES:
        return False
    raise ValueError('unrecognised boolean value: %r' % value)
```

It also makes it easier to see what the behaviour should be. e.g.

```python
# No default value, makes it more unclear what a user would expect. By default I would maybe expect an error.
str_to_bool('')

# seeing the default value makes it clear what the intended behaviour is
str_to_bool('', default_value=False)
```

Reply:
Maybe also worth noting, it appears that `get_env_var_or_use_default` seems to have a default value of `None` (unless specified). So the above might still break if the environment variable wasn't defined?